_posts/2021-01-15-docviewer-updates.md

```diff
@@ -0,0 +1,100 @@
+---
+title: "Insight to SNaaS delivery process"
+author: [pusztaienike]
+image: "../img/posts/docviewer_refactor.jpg"
+tags: [refactor, document viewer, saas, docviewer, document]
+redirect_to: https://www.sensenet.com/blog/2020-01-15-docviewer_updates
+---
+
+🌇 “Cities grow, cities evolve, cities have parts that simply die while other parts flourish; each city has to be renewed in order to meet the needs of its populace…” 🌆 -Grady Booch
+
+---
+
+The basic purpose of code refactoring is to make the code more efficient and maintainable. Of course we are about to keep our code up-to-date and optimize its performance so we thought that it is time to clean a lot of complicated lines of code in the document viewer. ♻️
+
+# Logic ✏️✂️🚀🐛👭
+
+The main goal was to simplify the code of the document viewer. There were some magic numbers, unnecessary props, useEffects that runs many times, so we had to go deep to rethink the logic. After planning, we had to rebuild the function based on the existing code. Although we try to avoid code repetition, sometimes it is even necessary to break down existing logic according to its function. As a result, Thumbnails and PageList became two separate components, for example. ✏️
+
+Not only the code has become simpler but also features. For example, rotating pages is only a form of visual representation, so it is unnecessary to store this information, it can be reproduced anytime. SearchBar widget become redundant too, because all browsers can search on page, so we removed this from selectable widgets. ✂️
```

Comment:
```suggestion
Not only the code has become simpler but also some of the features. For example, rotating pages is only a form of visual representation, so it is unnecessary to store this information, it can be reproduced anytime. SearchBar widget become redundant too, because all browsers can search on page, so we removed this from selectable widgets. ✂️
```